### PR TITLE
Eth2DeployBlockNumber function, update on Execution of Pruning

### PR DESCRIPTION
**Pull request by TKartist**
Eth2DeployBlockNumber function added to params.config file, and writeReceipt and ExecutionOfPrunningStage modified.

### CODE REVIEW

File: eth/stagedsync/stage_execute.go [hunk 1]
```diff
@@ -148,8 +148,10 @@ func executeBlock(
 	stateSyncReceipt = execRs.ReceiptForStorage
 
 	if writeReceipts {
-		if err = rawdb.AppendReceipts(tx, blockNum, receipts); err != nil {
-			return err
+		if params.Eth2DeployBlockNumber(cfg.chainConfig) >= blockNum {
```
Comment (AskAlexSharov) [inline eth/stagedsync/stage_execute.go:151]: Logic need to invert?

File: eth/stagedsync/stage_execute.go [hunk 2]
```diff
@@ -612,10 +614,13 @@ func PruneExecutionStage(s *PruneState, tx kv.RwTx, cfg ExecuteBlockCfg, ctx con
 			return err
 		}
 		// LogIndex.Prune will read everything what not pruned here
-		if err = rawdb.PruneTable(tx, kv.Log, cfg.prune.Receipts.PruneTo(s.ForwardProgress), ctx, math.MaxInt32); err != nil {
-			return err
+		if s.ForwardProgress > params.Eth2DeployBlockNumber(cfg.chainConfig) {
```
Comment (AskAlexSharov) [inline eth/stagedsync/stage_execute.go:617]: Logic need to invert?

File: params/config.go [hunk 3]
```diff
@@ -925,3 +925,22 @@ func NetworkIDByChainName(chain string) uint64 {
 		return config.ChainID.Uint64()
 	}
 }
+
+func Eth2DeployBlockNumber(c *ChainConfig) uint64 {
```
Comment (AskAlexSharov) [inline params/config.go:929]: Maybe calc this value once when app start and adjust cfg.prune.Receipts according to this numbers?

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]

File: eth/stagedsync/stage_execute.go [hunk 4]
```diff
@@ -588,6 +588,7 @@ func recoverCodeHashPlain(acc *accounts.Account, db kv.Tx, key []byte) {
 
 func PruneExecutionStage(s *PruneState, tx kv.RwTx, cfg ExecuteBlockCfg, ctx context.Context, initialCycle bool) (err error) {
 	logPrefix := s.LogPrefix()
+	depositeAddress := params.Eth2DeployBlockNumber(cfg.chainConfig)
```
Comment (AskAlexSharov) [inline eth/stagedsync/stage_execute.go:591]: I see on left side "Address" and on right side "blockNumber". Probably it's not address?

File: eth/stagedsync/stage_log_index.go [hunk 5]
```diff
@@ -75,7 +88,7 @@ func SpawnLogIndex(s *StageState, tx kv.RwTx, cfg LogIndexCfg, ctx context.Conte
 
 	startBlock := s.BlockNumber
 	pruneTo := cfg.prune.Receipts.PruneTo(endBlock)
-	if startBlock < pruneTo {
```
Comment (Giulio2002) [inline eth/stagedsync/stage_log_index.go:78]: just change the pruneTo value

File: eth/stagedsync/stage_log_index.go [hunk 6]
```diff
@@ -398,10 +411,23 @@ func PruneLogIndex(s *PruneState, tx kv.RwTx, cfg LogIndexCfg, ctx context.Conte
...
 		return err
 	}
+
+	chainConfig, err := rawdb.ReadChainConfig(tx, hash)
+	if err != nil {
+		return err
+	}
+
+	blockNum := params.Eth2DeployBlockNumber(chainConfig)
+	pruneTo := cfg.prune.Receipts.PruneTo(s.ForwardProgress)
```
Comment (Giulio2002) [inline eth/stagedsync/stage_log_index.go:425]: just change the pruneTo value

**Review by Giulio2002 — CHANGES_REQUESTED** [review]

File: params/config.go (hunk 3 above)
Comment (Giulio2002) [inline params/config.go:929]: return uint64 as a pointer and return nil if it does not exist for a specific chain config

**Review by Giulio2002 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**